### PR TITLE
Issue 1827: Add demographics to tiles and download files

### PR DESCRIPTION
**Pull request by lucasmbrown-usds**
Addresses #1827.

### CODE REVIEW

File: data/data-pipeline/data_pipeline/content/config/csv.yml [hunk 1]
```diff
@@ -14,6 +14,30 @@ fields:
   - score_name: State/Territory
     label: State/Territory
     format: string
+  - score_name: Percent Black or African Amer.
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/content/config/csv.yml:17]: nit: can we spell this out?

File: data/data-pipeline/data_pipeline/content/config/csv.yml [hunk 2]
```diff
@@ -14,6 +14,30 @@ fields:
...
+    label: Percent Black or African American alone
+    format: float
+  - score_name: Percent Amer. Indian / Alaska Native
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/content/config/csv.yml:20]: Same nit

File: data/data-pipeline/data_pipeline/content/config/csv.yml [hunk 3]
```diff
@@ -14,6 +14,30 @@ fields:
...
+  - score_name: Percent Two or more
+    label: Percent Two or more
+    format: float
+  - score_name: Percent White
+    label: Percent White
+    format: float
+  - score_name: Percent Hispanic or Latino
+    label: Percent Hispanic or Latino
+    format: float
+  - score_name: Percent Other
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/content/config/csv.yml:38]: nit: this won't be super clear in the download -- percent other race or ethnicity might be clearer?

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py [hunk 4]
```diff
@@ -405,36 +398,42 @@ def transform(self) -> None:
...
 
         # Calculate some demographic information.
-        df[self.BLACK_FIELD_NAME] = df["B02001_003E"]
-        df[self.AMERICAN_INDIAN_FIELD_NAME] = df["B02001_004E"]
-        df[self.ASIAN_FIELD_NAME] = df["B02001_005E"]
-        df[self.HAWAIIAN_FIELD_NAME] = df["B02001_006E"]
-        df[self.TWO_OR_MORE_RACES_FIELD_NAME] = df["B02001_008E"]
-        df[self.NON_HISPANIC_WHITE_FIELD_NAME] = df["B03002_003E"]
-        df[self.HISPANIC_FIELD_NAME] = df["B03003_003E"]
+        df[field_names.BLACK_FIELD_NAME] = df["B02001_003E"]
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:401]: Can we change this to a rename?

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py [hunk 5]
```diff
@@ -405,36 +398,42 @@ def transform(self) -> None:
...
+        df[field_names.ASIAN_FIELD_NAME] = df["B02001_005E"]
+        df[field_names.HAWAIIAN_FIELD_NAME] = df["B02001_006E"]
+        df[field_names.TWO_OR_MORE_RACES_FIELD_NAME] = df["B02001_008E"]
+        df[field_names.NON_HISPANIC_WHITE_FIELD_NAME] = df["B03002_003E"]
+        df[field_names.HISPANIC_FIELD_NAME] = df["B03003_003E"]
+        df[field_names.OTHER_RACE_FIELD_NAME] = df["B02001_007E"]
 
         # Calculate demographics as percent
-        df[self.PERCENT_PREFIX + self.BLACK_FIELD_NAME] = (
+        df[field_names.PERCENT_PREFIX + field_names.BLACK_FIELD_NAME] = (
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:411]: Can we use the fields defined above here?

Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:411]: Or better yet, collapse it as:

```
for c in race_cols:
    df[prefix + col] = df[col] / total_population_series
```

File: data/data-pipeline/data_pipeline/score/field_names.py [hunk 6]
```diff
@@ -91,6 +91,32 @@
     "Low median household income as a percent of area median income"
 )
 
+# Additional ACS demographic fields.
+BLACK_FIELD_NAME = "Black or African Amer."
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/score/field_names.py:95]: See notes above but also ty for pulling this out to field names!

We also might not need the population count fields, and can keep just the percentage fields. Wdyt? The total_population + percentages should be comprehensive.

File: data/data-pipeline/data_pipeline/tests/sources/eamlis/test_etl.py [hunk 7]
```diff
@@ -63,7 +63,7 @@ def setup_method(self, _method, filename=__file__):
     def test_init(self, mock_etl, mock_paths):
         """Tests that the mock NationalRiskIndexETL class instance was
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/tests/sources/eamlis/test_etl.py:64]: `NationalRiskIndexETL` seems like we should remove it?

**Review by emma-nechamkin — COMMENTED** [review]
I had mostly nits and stylistic comments! I think this code is pretty verbose (everything is lol) and it might be a bit clearer if we iterate rather than doing some operations explicitly.

File: data/data-pipeline/data_pipeline/content/config/csv.yml (hunk 1 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/content/config/csv.yml:17]: I'm just following @KameronKerger 's field names!

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py (hunk 4 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:401]: good point! will do. leave it better than you found it.

File: data/data-pipeline/data_pipeline/score/field_names.py (hunk 6 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/score/field_names.py:97]: Oooh good point. I'll change that.

File: data/data-pipeline/data_pipeline/tests/sources/eamlis/test_etl.py (hunk 7 above)
Comment (lucasmbrown-usds) [reply data/data-pipeline/data_pipeline/tests/sources/eamlis/test_etl.py:64]: Yeah this isn't my code -- if you fix that on your branch, great! 

I ran `black` on all .py files and it cleaned up this one.

**Comment by lucasmbrown-usds** [comment]
Tasks: 

- [ ] Fix tests 
- [ ] Update field names after confirming with Kameron 
- [ ] Refactor ACS ETL to use `renames` 
- [ ] Collapse percent demographics
- [ ] Drop population count fields and just keep the percentage fields

**Comment by lucasmbrown-usds** [comment]
Because updating the test pickles is complicated and involves a lot of merge conflicts, I went ahead and included the three age buckets in this pull request as well. 

So now the pull request is beefier and is slightly more complicated to review, but this feels like the right tradeoff. 

Feedback welcome!

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py [hunk 8]
```diff
@@ -186,19 +186,25 @@ def __init__(self):
...
-        )
-        self.ASIAN_FIELD_NAME = "Asian alone"
-        self.HAWAIIAN_FIELD_NAME = "Native Hawaiian and Other Pacific alone"
-        self.TWO_OR_MORE_RACES_FIELD_NAME = "Two or more races"
-        self.NON_HISPANIC_WHITE_FIELD_NAME = "Non-Hispanic White"
+        self.BLACK_FIELD_NAME = "Black or African American"
+        self.AMERICAN_INDIAN_FIELD_NAME = "American Indian / Alaska Native"
+        self.ASIAN_FIELD_NAME = "Asian"
+        self.HAWAIIAN_FIELD_NAME = "Native Hawaiian or Pacific"
+        self.TWO_OR_MORE_RACES_FIELD_NAME = "two or more races"
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:196]: Please capitalize `Two or more races`

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py [hunk 9]
```diff
@@ -186,19 +186,25 @@ def __init__(self):
...
+        self.NON_HISPANIC_WHITE_FIELD_NAME = "White"
         self.HISPANIC_FIELD_NAME = "Hispanic or Latino"
+        # Note that `other` is lowercase because the whole field will show up in the download
+        # file as "Percent other races"
+        self.OTHER_RACE_FIELD_NAME = "other races"
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:201]: `other race` (I think "other races" seems a bit too much like a multiracial descriptor, but it's not

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py (hunk 8 above)
Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:196]: Does the whole field show up as per below?

Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:196]: Yes! ignore

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py (hunk 9 above)
Comment (emma-nechamkin) [reply data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:201]: Eh, this is purely my personal preference, I'm gonna withdraw it

File: data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py [hunk 10]
```diff
@@ -406,36 +472,103 @@ def transform(self) -> None:
...
+                "B02001_008E": self.TWO_OR_MORE_RACES_FIELD_NAME,
+                "B03002_003E": self.NON_HISPANIC_WHITE_FIELD_NAME,
+                "B03003_003E": self.HISPANIC_FIELD_NAME,
+                "B02001_007E": self.OTHER_RACE_FIELD_NAME,
+                "B02001_001E": self.TOTAL_RACE_POPULATION_FIELD_NAME,
+            },
+            errors="raise",
+        )
+
+        for race_field_name in self.RE_OUTPUT_FIELDS:
```
Comment (emma-nechamkin) [inline data/data-pipeline/data_pipeline/etl/sources/census_acs/etl.py:491]: nice!

**Review by emma-nechamkin — APPROVED** [review]
Nice!

**Pull request merged**